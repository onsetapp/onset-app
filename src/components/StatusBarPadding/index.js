import React, { Component } from 'react'
import { View, Platform, StyleSheet } from 'react-native'
import Constants from 'expo-constants'
import PropTypes from 'prop-types'

export default class StatusBarPadding extends Component {
  static propTypes = {
    children: PropTypes.node
  }

  static defaultProps = {}

  render () {
    return (
      <View style={ styles.statusBarHeight }>
        { this.props.children }
      </View>
    )
  }
}

const styles = StyleSheet.create({
  statusBarPadding: {
    flex: 1,
    paddingTop: Platform.OS === 'ios' ? 0 : Constants.statusBarHeight,
  }
})
