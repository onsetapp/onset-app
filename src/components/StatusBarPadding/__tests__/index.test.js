import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import StatusBarPadding from '..'

describe('<StatusBarPadding />', () => {
  let component
  beforeEach(() => {
    component = shallow(<StatusBarPadding />)
  })

  it('renders the correct text', () => {
    expect(true).toBeTruthy()
    expect(component).toBeTruthy()
  })

  it('renders correctly', () => {
    const tree = renderer.create(<StatusBarPadding />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
