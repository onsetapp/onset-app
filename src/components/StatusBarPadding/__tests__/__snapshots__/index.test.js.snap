// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusBarPadding /> renders correctly 1`] = `<View />`;
